### PR TITLE
EVG-5685 Create new host allocation job

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -61,32 +63,89 @@ func (s *distroScheduler) scheduleDistro(distroID string, runnableTasksForDistro
...
+			name = task.GetTaskGroupString()
+		}
+
+		duration := task.FetchExpectedDuration()
+		task.ExpectedDuration = duration
+		distroExpectedDuration += duration
+
+		var taskGroupInfo model.TaskGroupInfo
+		if info, exists := taskGroupInfosMap[name]; exists {
+			info.Count++
```
[the remark this answers is not in the document]
Comment (bsamek) [reply scheduler/scheduler.go]: Nevermind, I missed the assignment.

File: scheduler/task_queue_persister.go [hunk 2]
```diff
@@ -10,15 +10,16 @@ import (
 // and saving it.
 type TaskQueuePersister interface {
 	// distro, tasks, duration cache
-	PersistTaskQueue(string, []task.Task) ([]model.TaskQueueItem, error)
+	// Do we actually need to return []model.TaskQueueItem?
```
[the remark this answers is not in the document]
Comment (StuartButler) [reply scheduler/task_queue_persister.go]: Oops - yes, it is.

File: model/task_queue.go [hunk 3]
```diff
@@ -32,18 +32,29 @@ type TaskGroupInfo struct {
...
+	MaxDurationThreshold time.Duration   `bson:"max_duration_threshold" json:"max_duration_threshold"`
+	CountOverThreshold   int             `bson:"count_over_threshold" json:"count_over_threshold"`
+	TaskGroupInfos       []TaskGroupInfo `bson:"task_group_infos" json:"task_group_infos"`
+}
+
+func GetDistroQueueInfo(distroID string) (DistroQueueInfo, error) {
+	taskQueue := &TaskQueue{}
+	query := bson.M{taskQueueDistroKey: distroID}
+	projection := bson.M{taskQueueDistroQueueInfoKey: 1, taskQueueIdKey: 0}
+	err := db.FindOne(TaskQueuesCollection, query, projection, db.NoSort, taskQueue)
```
Comment (tychoish) [inline model/task_queue.go]: nit: combine lines 44-46 and wrap the error.

Comment (StuartButler) [reply model/task_queue.go]: Done

**Pull request merged**